src/main/java/uk/nhs/hee/tis/trainee/credentials/service/CachingDelegate.java

```diff
@@ -247,24 +247,26 @@ public Optional<String> getUnverifiedSessionIdentifier(UUID key) {
   }
 
   /**
-   * Cache a verified session identifier for later retrieval.
+   * Cache a verified session's identity unique identifier for later retrieval.
    *
-   * @param sessionId The identifier of the verified session to cache.
+   * @param sessionId  The identifier of the verified session to cache.
+   * @param identityId The verified identity's unique identifier.
    * @return The cached session identifier.
    */
-  @CachePut(cacheNames = VERIFIED_SESSION_IDENTIFIER, cacheManager = VERIFIED_SESSION_DATA)
-  public String cacheVerifiedSessionIdentifier(String sessionId) {
-    return sessionId;
+  @CachePut(cacheNames = VERIFIED_SESSION_IDENTIFIER,
+      cacheManager = VERIFIED_SESSION_DATA, key = "#sessionId")
+  public UUID cacheVerifiedIdentityIdentifier(String sessionId, UUID identityId) {
```

Comment:
Do we need to cache the sessionId in case a single user has more than one session open at a time? I was initially wondering if we needed it any more...

Reply:
The "sessionId" is really just a generated identifier from the Cognito token, and is stable for a whole login (included across token refreshes). Caching against it allows us to track the verified identity, while invalidating the session upon sign out.
It's built that way as an alternative to using a server session, which were a bit fiddly given our requirements - though I can't remember the specifics.

Reply:
Ah cool, I was forgetting the cognito token! 👍